allinone/Dockerfile

```diff
@@ -1,62 +1,129 @@
-FROM jenkinsci/jnlp-slave:3.10-1
+FROM jenkinsci/jnlp-slave:3.23-1
 
 USER root
 
 WORKDIR /opt
 
 # Add docker client
 # The version number is the docker version you want to move to
-ENV DOCKER_VERION 17.05.0-ce
-RUN wget https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_VERION}.tgz
-RUN tar -zxvf docker-${DOCKER_VERION}.tgz
-RUN cp docker/docker /usr/bin/
-RUN rm -rf /opt/docker*
+ENV DOCKER_VERION 17.09.0-ce
+RUN curl -fsSL get.docker.com -o get-docker.sh
+RUN sh get-docker.sh
+RUN usermod -aG docker jenkins
+RUN usermod -aG docker root
 
 # Add kubectl client
-ENV KUBECTL_VERSION 1.7.0
-RUN wget https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl
-RUN chmod +x kubectl
-RUN mv kubectl /usr/bin/
+#    ENV KUBECTL_VERSION 1.11.1
+#RUN wget https://storage.googleapis.com/kubernetes-release/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl
+#RUN chmod +x kubectl
+#RUN mv kubectl /usr/bin/
```

Comment:
It looks like we are removing functionality from this `allinone` jenkins slave.  We shouldnt remove functionality.  It looks like you only want maven in the jenkins slave?  We should just make another slave named `maven`.